src/shared/views/Mapper/CardBody.js

```diff
@@ -54,125 +51,138 @@ const tutorialModes = {
-                const tileMaxX = tileMinX + tilesPerRow;
-                for (let tileX = tileMinX; tileX < tileMaxX; tileX += 1) {
-                    const taskIdx = group.tasks.findIndex(
-                        (e) =>
-                            parseInt(e.taskX, 10) === tileX &&
-                            parseInt(e.taskY, 10) === tileY,
-                    );
-                    if (taskIdx > -1) {
-                        // we have a valid task for these coordinates
-                        cardToPush.validTiles += 1;
-                        const tile = group.tasks[taskIdx];
-                        tileRowObject.tiles.push(tile);
-                        // store a 0 result for each tile
-                        onToggleTile({
-                            resultId: tile.taskId,
-                            result: 0,
-                            groupId: tile.groupId,
-                            projectId: tile.projectId,
-                        });
-                    } else {
-                        // no task: insert an empty tile marker
-                        tileRowObject.tiles.push('emptytile');
-                    }
+    generateTasks = () => {
+        // build an array of tasks grouped by 6 so that each
+        // item in the array holds all the info for 1 screen
+        // this array will then be passed to the FlatList.data prop
+        // and each item will be passed to FlatList.renderItem()
+        const {
+            group: { groupId, projectId, xMax, xMin, yMax, yMin, tasks },
+            tileServer,
+            tileServerB,
+            tutorial,
+            zoomLevel,
+        } = this.props;
+        const minx = parseInt(xMin, 10);
+        const maxx = parseInt(xMax, 10);
+        const miny = parseInt(yMin, 10);
+        const maxy = parseInt(yMax, 10);
+        // screens contains items of 6 elements, organized in columns,
+        // sorted by X then Y, so that the first 3 items are 1 column, the
+        // next 3 are the next column to the right of it, etc...
+        const numScreens = Math.ceil((1 + maxx - minx) / 2);
+        const screens = [...Array(numScreens)].map(() => Array(6));
 
-                    if (tileY > tileRowObject.rowYEnd) {
-                        tileRowObject.rowYEnd = tileY;
-                    }
-                    if (tileX > tileRowObject.cardXEnd) {
-                        tileRowObject.cardXEnd = tileX;
-                    }
+        if (!tutorial) {
+            // in "real" mapping sessions, we don't download tasks from the server,
+            // instead we create them from data in the group object here
+            for (let x = minx; x <= maxx; x += 1) {
+                for (let y = miny; y <= maxy; y += 1) {
+                    const dX = x - minx;
+                    const screen = Math.floor(dX / 2);
+                    const column = dX % 2;
+                    const row = y - miny;
+                    const urlB = tileServerB
```

Comment:
why is `urlB` defined here? It is also defined as an attribute of `task` below. Or the calculation could be moved some lines below...